modules/apps/object/object-api/src/main/java/com/liferay/object/field/util/ObjectFieldUtil.java

```diff
@@ -67,16 +67,41 @@ public static ObjectField createObjectField(
 		return createObjectField(businessType, dbType, name, name, false);
 	}
 
+	public static ObjectField createObjectField(
+		String businessType, String dbType, String name,
+		List<ObjectFieldSetting> objectFieldSettings) {
+
+		return createObjectField(
+			businessType, null, dbType, null, false, null, name, 0, name,
+			objectFieldSettings, false, false, false);
+	}
+
+	public static ObjectField createObjectField(
+		String businessType, String dbType, String label, String name) {
+
+		return createObjectField(businessType, dbType, label, name, false);
+	}
+
+	public static ObjectField createObjectField(
+		String businessType, String dbType, String label, String name,
+		boolean required) {
+
+		return createObjectField(
+			0, businessType, null, dbType, null, false, null, label, name,
+			required, false, false);
+	}
+
 	public static ObjectField createObjectField(
 		String businessType, String dbColumnName, String dbType,
-		boolean indexed, boolean indexedAsKeyword, String indexedLanguageId,
-		String label, long listTypeDefinitionId, String name,
-		List<ObjectFieldSetting> objectFieldSettings, boolean required,
-		boolean system) {
+		String externalReferenceCode, boolean indexedAsKeyword,
+		String indexedLanguageId, String label, long listTypeDefinitionId,
+		String name, List<ObjectFieldSetting> objectFieldSettings,
+		boolean required, boolean system, boolean indexed) {
```

Comment:
@seltong please reorder so that `externalReferenceCode` comes before the `businessType`, and  do not change the order of the `indexed` property.. We should follow the same order from the service.xml